src/runtime.js

```diff
@@ -196,6 +196,11 @@ var Runtime = {
 #if ASSERTIONS
       assert(args.length == sig.length-1);
 #endif
+      if (Module['usingWasm']) {
+        // look up function by name and call it directly
+        name = Module['indirectFunctions'][ptr]
+        return Module[name].apply(null, args);
```

Comment:
I guess it would also be valid here to do
`Module['asm'].apply(null, args)` which would avoid looking up the function name in the module.